### PR TITLE
fix: nested layout min width limit on Lg screen

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**
---

> [!IMPORTANT]
> Please, **don't forget to run `rush change`** for the commits that introduce **new features** 🙏

---

Refer to [documentation](https://github.com/gooddata/gooddata-ui-sdk/blob/master/dev_docs/continuous_integration.md) to see how to run checks and tests in the pull request. This is the list of the most used commands:

```
extended test - backstop
```

```
extended test - tiger-cypress - integrated
extended test - tiger-cypress - isolated
extended test - tiger-cypress - record
```

### CODE REVIEW

File: libs/sdk-ui-dashboard/src/_staging/layout/sizing.ts [hunk 1]
```diff
@@ -352,3 +356,142 @@ export const getDashboardLayoutItemHeight = (size: IDashboardLayoutSize): number
 
 export const getDashboardLayoutItemHeightForGrid = (gridHeight: number): number =>
     gridHeight * GRID_ROW_HEIGHT_IN_PX;
+
+export const determineSizeForScreen = (
+    screen: ScreenSize,
+    layoutItemSize?: IDashboardLayoutSizeByScreenSize,
+): IDashboardLayoutSize => {
+    return {
+        ...(layoutItemSize ? layoutItemSize[screen] : {}),
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/_staging/layout/sizing.ts:365]: there could still be a null pointer exc when `layoutItemSize` does not have size for provided screen. This should handle it: 
`...(layoutItemSize ? layoutItemSize[screen] ?? {} : {}),`

File: libs/sdk-ui-dashboard/src/_staging/layout/sizing.ts [hunk 2]
```diff
@@ -352,3 +356,142 @@ export const getDashboardLayoutItemHeight = (size: IDashboardLayoutSize): number
...
+        gridWidth: determineWidthForScreen(screen, layoutItemSize),
+    };
+};
+
+export const determineWidthForScreen = (
+    screen: ScreenSize,
+    layoutItemSize?: IDashboardLayoutSizeByScreenSize,
+) => {
+    // Determine if element has size set in metadata object for the current screen size
+    const providedSizeForScreen = layoutItemSize ? layoutItemSize[screen] : undefined;
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/_staging/layout/sizing.ts:375]: `const providedSizeForScreen = layoutItemSize.?[screen];` is shorter (this is probably older code)

File: libs/sdk-ui-dashboard/src/model/store/index.ts [hunk 3]
```diff
@@ -181,9 +181,12 @@ export { selectLegacyDashboards } from "./legacyDashboards/legacyDashboardsSelec
 
 export type { UndoEnhancedState, UndoEntry } from "./_infra/undoEnhancer.js";
 export type { LayoutState, LayoutStash } from "./layout/layoutState.js";
+
+export { layoutActions } from "./layout/index.js";
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/model/store/index.ts:185]: Should this be exposed? I don't see other store actions being exposed directly.

File: libs/sdk-ui-dashboard/src/presentation/dashboard/components/DashboardScreenSizeContext.tsx [hunk 4]
```diff
@@ -25,6 +37,20 @@ const getScreenSize = (ref: React.RefObject<HTMLDivElement>): ScreenSize => {
     return "xs";
 };
 
+function setNewSize(
+    screenSize: string | undefined,
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/presentation/dashboard/components/DashboardScreenSizeContext.tsx:41]: maybe rename this to `previousScreenSize`?

File: libs/sdk-ui-dashboard/src/presentation/dashboard/components/DashboardScreenSizeContext.tsx [hunk 5]
```diff
@@ -25,6 +37,20 @@ const getScreenSize = (ref: React.RefObject<HTMLDivElement>): ScreenSize => {
...
+    setScreenSize: React.Dispatch<React.SetStateAction<ScreenSize | undefined>>,
+    dispatch: React.Dispatch<AnyAction>,
+) {
+    const newSize = getScreenSize();
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/presentation/dashboard/components/DashboardScreenSizeContext.tsx:45]: possibly change this to `const currentScreenSize = detectCurrentScreenSize();` or `getCurrentScreenSize`

File: libs/sdk-ui-dashboard/src/_staging/layout/sizing.ts (hunk 1 above)
Comment (ivan-nejezchleb) [reply libs/sdk-ui-dashboard/src/_staging/layout/sizing.ts:365]: fixed

File: libs/sdk-ui-dashboard/src/_staging/layout/sizing.ts (hunk 2 above)
Comment (ivan-nejezchleb) [reply libs/sdk-ui-dashboard/src/_staging/layout/sizing.ts:375]: fixed

File: libs/sdk-ui-dashboard/src/presentation/dashboard/components/DashboardScreenSizeContext.tsx [hunk 6]
```diff
@@ -33,21 +59,29 @@ const DashboardScreenSizeContext = React.createContext<ScreenSize | undefined>(u
...
         };
-    }, [layoutRef]);
+    }, [screenSize, dispatch]);
 
     return (
-        <div ref={layoutRef}>
+        <div className="s-screen-size-container">
             <DashboardScreenSizeContext.Provider value={screenSize}>
-                {children}
+                {!appStateScreenSize ? null : children}
```
[the remark this answers is not in the document]
Comment (ivan-nejezchleb) [reply libs/sdk-ui-dashboard/src/presentation/dashboard/components/DashboardScreenSizeContext.tsx:84]: fixed

File: libs/sdk-ui-dashboard/src/presentation/dashboard/components/DashboardScreenSizeContext.tsx (hunk 5 above)
Comment (ivan-nejezchleb) [reply libs/sdk-ui-dashboard/src/presentation/dashboard/components/DashboardScreenSizeContext.tsx:45]: fixed

File: libs/sdk-ui-dashboard/src/model/store/index.ts (hunk 3 above)
Comment (ivan-nejezchleb) [reply libs/sdk-ui-dashboard/src/model/store/index.ts]: replaced by command+event combo

**Review by xMort — APPROVED** [review]

**Review by xMort — APPROVED** [review]

**Pull request merged**